CAAppKitBridge/Source/NSView+CAMethods.m

```diff
@@ -0,0 +1,201 @@
+   Boston, MA 02110-1301, USA.
+*/
+
+#import "GSCAData.h"
+#import "NSView+CAMethods.h"
+#import <Foundation/Foundation.h>
+
+@implementation NSView (NSViewCAmethods)
+- (BOOL) wantsLayer {
+  if (self->_coreAnimationData == nil) {
+      return NO;
+  }
+  GSCAData * GSCAData = self->_coreAnimationData;
+  return GSCAData->_wantsLayer;
+}
+
+- (void) setWantsLayer:(BOOL) newValue{
+  if (newValue == NO){
+      return;
+  }
+
+  /* Initialise new GSCAData if setWantsLayer:YES */
+  GSCAData * currGSCAData = [[GSCAData alloc]init];
+  currGSCAData->_wantsLayer = YES;
+  currGSCAData->_isRootLayer = YES;
+  currGSCAData->_layer = [self makeBackingLayer];
+  //GSCAData->_renderer = 
+  //    [CARenderer rendererWithNSOpenGLContext: [self openGLContext].CGLContextObj
+  //                                                   options: nil];
+  /* Attach GSCAData to self */
+  self->_coreAnimationData = currGSCAData;
+
+  /* Further prep of CARenderer */
+  [currGSCAData->_renderer setLayer: currGSCAData->_layer];           // Set root layer
+  [currGSCAData->_renderer setBounds: NSRectToCGRect([self bounds])]; // Set bounds
+  [self openGLContext];                                               // Create OpenGL context
+
+  /* Call _recursiveSubviewPropagation recursively on all the subviews */
+  for (NSView *currView in [self subviews])
+  {
+      [currView _recursiveSubviewPropagation];
+  }
+}
+
+- (void) _recursiveSubviewPropagation {
+  /* Initialise new GSCAData instance */
+  GSCAData * currGSCAData = [[GSCAData alloc]init];
+  currGSCAData->_wantsLayer = NO;
+  currGSCAData->_isRootLayer = NO;
+  currGSCAData->_layer = [self makeBackingLayer];
+
+  /* Attach GSCAData to self */
+  self->_coreAnimationData = currGSCAData;
+
+  /* Attach our CALayer to its superView CALayer */
+  NSView * superView = [self superview];
+  if(superView != nil){
+      GSCAData * superGSCAData = superView->_coreAnimationData;
+      [superGSCAData->_layer addSublayer:currGSCAData->_layer];
+  }
+
+  /* Call wantsLayer recursively on all the subviews */
+  for (NSView *currView in [self subviews])
+  {
+      [currView _recursiveSubviewPropagation];
+  }
+
+}
+
+- (BOOL) addCARenderer: (CARenderer*) customCARenderer {
+  GSCAData *currGSCAData = self->_coreAnimationData;
+  if (!currGSCAData->_isRootLayer) {
+      NSLog(@"Cannot add CARenderer to a non-root layer");
+      return NO;
+  }
+  currGSCAData->_renderer = customCARenderer;
+  return YES;
+}
+
+- (BOOL) removeCARenderer {
+  GSCAData *currGSCAData = self->_coreAnimationData;
+  if (!currGSCAData->_isRootLayer) {
+      NSLog(@"Cannot remove CARenderer from a non-root layer");
+      return NO;
+  }
+  currGSCAData->_renderer = nil;
+  return YES;
+}
+
+- (CALayer *) makeBackingLayer {
+  return [CALayer layer];
+}
+
+
+/* methods from libs-gui/Headers/AppKit/NSOpenGlView.h */
+
+
+/**
+ detach from the current context.  You should call it before releasing this 
+ object.
+*/
+- (void) clearGLContext
```

Comment:
If you need these, they need to be renamed not to conflict. I recall we agreed you don't need them.